File: frontend/src/services/parameter.ts
/**
 * 参数化配置服务
 */

import api from './api'
import type { ApiResponse } from '@/types/api'

// 参数类型
export interface ParameterConfig {
  id: number
  name: string
  param_type: string
  static_value: string | null
  config: string | null
  description: string | null
  is_enabled: boolean
  sort_order: number
  created_at: string | null
  updated_at: string | null
}

export interface ParameterConfigCreate {
  name: string
  param_type: string
  static_value?: string | null
  config?: string | null
  description?: string | null
  is_enabled?: boolean
  sort_order?: number
}

export interface ParameterListResponse {
  items: ParameterConfig[]
  total: number
}

// 参数类型映射
export const PARAM_TYPES: Record<string, string> = {
  static: '静态值',
  timestamp: '时间戳',
  random: '随机数',
  uuid: 'UUID',
  counter: '计数器',
  custom: '自定义函数',
}

// 内置参数列表
export const BUILTIN_PARAMS = [
  { name: 'timestamp', description: '当前时间戳（毫秒）', example: '${timestamp}' },
  { name: 'timestamp_s', description: '当前时间戳（秒）', example: '${timestamp_s}' },
  { name: 'datetime', description: '当前日期时间', example: '${datetime}' },
  { name: 'date', description: '当前日期', example: '${date}' },
  { name: 'time', description: '当前时间', example: '${time}' },
  { name: 'uuid', description: 'UUID', example: '${uuid}' },
  { name: 'uuid_short', description: 'UUID（无横线）', example: '${uuid_short}' },
  { name: 'random_int', description: '随机整数', example: '${random_int}' },
  { name: 'random_str', description: '随机字符串', example: '${random_str}' },
]

class ParameterService {
  private baseUrl = '/parameters'

  /**
   * 获取参数类型列表
   */
  async getTypes(): Promise<ApiResponse<Record<string, string>>> {
    return api.get(`${this.baseUrl}/types`)
  }

  /**
   * 获取参数配置列表
   */
  async getList(params?: {
    page?: number
    page_size?: number
    name?: string
    param_type?: string
    is_enabled?: boolean
  }): Promise<ApiResponse<ParameterListResponse>> {
    return api.get(this.baseUrl, { params })
  }

  /**
   * 获取所有启用的参数配置
   */
  async getEnabled(): Promise<ApiResponse<ParameterConfig[]>> {
    return api.get(`${this.baseUrl}/enabled`)
  }

  /**
   * 获取单个参数配置
   */
  async getById(id: number): Promise<ApiResponse<ParameterConfig>> {
    return api.get(`${this.baseUrl}/${id}`)
  }

  /**
   * 创建参数配置
   */
  async create(data: ParameterConfigCreate): Promise<ApiResponse<ParameterConfig>> {
    return api.post(this.baseUrl, data)
  }

  /**
   * 更新参数配置
   */
  async update(id: number, data: Partial<ParameterConfigCreate>): Promise<ApiResponse<ParameterConfig>> {
    return api.put(`${this.baseUrl}/${id}`, data)
  }

  /**
   * 删除参数配置
   */
  async delete(id: number): Promise<ApiResponse<boolean>> {
    return api.delete(`${this.baseUrl}/${id}`)
  }

  /**
   * 预览参数替换结果
   */
  async preview(content: string): Promise<ApiResponse<string>> {
    return api.post(`${this.baseUrl}/preview`, null, { params: { content } })
  }
}

export const parameterService = new ParameterService()

File: backend/src/schemas/parameter.py
"""
参数化配置 Schema
"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class ParameterConfigBase(BaseModel):
    """参数化配置基类"""

    name: str = Field(..., min_length=1, max_length=100, description="参数名称")
    param_type: str = Field(default="static", description="参数类型")
    static_value: Optional[str] = Field(None, description="静态值或默认值")
    config: Optional[str] = Field(None, description="参数配置(JSON格式)")
    description: Optional[str] = Field(None, max_length=500, description="参数描述")
    is_enabled: bool = Field(default=True, description="是否启用")
    sort_order: int = Field(default=0, description="排序")


class ParameterConfigCreate(ParameterConfigBase):
    """创建参数化配置"""

    pass


class ParameterConfigUpdate(BaseModel):
    """更新参数化配置"""

    name: Optional[str] = Field(None, min_length=1, max_length=100, description="参数名称")
    param_type: Optional[str] = Field(None, description="参数类型")
    static_value: Optional[str] = Field(None, description="静态值或默认值")
    config: Optional[str] = Field(None, description="参数配置(JSON格式)")
    description: Optional[str] = Field(None, max_length=500, description="参数描述")
    is_enabled: Optional[bool] = Field(None, description="是否启用")
    sort_order: Optional[int] = Field(None, description="排序")


class ParameterConfigResponse(ParameterConfigBase):
    """参数化配置响应"""

    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ParameterConfigListResponse(BaseModel):
    """参数化配置列表响应"""

    items: List[ParameterConfigResponse]
    total: int


# 参数类型枚举
PARAM_TYPES = {
    "static": "静态值",
    "timestamp": "时间戳",
    "random": "随机数",
    "uuid": "UUID",
    "counter": "计数器",
    "custom": "自定义函数",
}